x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/utils.test.ts

```diff
@@ -68,39 +73,103 @@ describe('crawlerDomainServerToClient', () => {
+    expect(
+      crawlRequestServerToClient({
+        ...defaultServerPayload,
+        completed_at: 'Mon, 31 Aug 2020 17:00:00 +0000',
+      })
+    ).toStrictEqual({ ...defaultClientPayload, completedAt: 'Mon, 31 Aug 2020 17:00:00 +0000' });
+  });
+});
+
 describe('crawlerDataServerToClient', () => {
-  it('converts all domains from the server form to their client form', () => {
-    const domains: CrawlerDomainFromServer[] = [
-      {
-        id: 'x',
-        name: 'moviedatabase.com',
-        document_count: 13,
-        created_on: 'Mon, 31 Aug 2020 17:00:00 +0000',
-        sitemaps: [],
-        entry_points: [],
-        crawl_rules: [],
-        default_crawl_rule: DEFAULT_CRAWL_RULE,
-      },
-      {
-        id: 'y',
-        name: 'swiftype.com',
-        last_visited_at: 'Mon, 31 Aug 2020 17:00:00 +0000',
-        document_count: 40,
-        created_on: 'Mon, 31 Aug 2020 17:00:00 +0000',
-        sitemaps: [],
-        entry_points: [],
-        crawl_rules: [],
-      },
-    ];
-
-    const output = crawlerDataServerToClient({
+  let output: CrawlerData;
+
+  const domains: CrawlerDomainFromServer[] = [
+    {
+      id: 'x',
+      name: 'moviedatabase.com',
+      document_count: 13,
+      created_on: 'Mon, 31 Aug 2020 17:00:00 +0000',
+      sitemaps: [],
+      entry_points: [],
+      crawl_rules: [],
+      default_crawl_rule: DEFAULT_CRAWL_RULE,
+    },
+    {
+      id: 'y',
+      name: 'swiftype.com',
+      last_visited_at: 'Mon, 31 Aug 2020 17:00:00 +0000',
+      document_count: 40,
+      created_on: 'Mon, 31 Aug 2020 17:00:00 +0000',
+      sitemaps: [],
+      entry_points: [],
+      crawl_rules: [],
+    },
+  ];
+
+  const crawlRequests: CrawlRequestFromServer[] = [
+    {
+      id: 'a',
+      status: CrawlerStatus.Canceled,
+      created_at: 'Mon, 31 Aug 2020 11:00:00 +0000',
+      began_at: 'Mon, 31 Aug 2020 12:00:00 +0000',
+      completed_at: 'Mon, 31 Aug 2020 13:00:00 +0000',
+    },
+    {
+      id: 'b',
+      status: CrawlerStatus.Success,
+      created_at: 'Mon, 31 Aug 2020 14:00:00 +0000',
+      began_at: 'Mon, 31 Aug 2020 15:00:00 +0000',
+      completed_at: 'Mon, 31 Aug 2020 16:00:00 +0000',
+    },
+  ];
+
+  beforeAll(() => {
+    output = crawlerDataServerToClient({
       domains,
+      crawl_requests: crawlRequests,
     });
+  });
 
+  it('converts all domains from the server form to their client form', () => {
     expect(output.domains).toHaveLength(2);
     expect(output.domains[0]).toEqual(crawlerDomainServerToClient(domains[0]));
     expect(output.domains[1]).toEqual(crawlerDomainServerToClient(domains[1]));
   });
+
+  it('converts all crawl requests from the server form to their client form', () => {
+    expect(output.crawlRequests).toHaveLength(2);
+    expect(output.crawlRequests[0]).toEqual(crawlRequestServerToClient(crawlRequests[0]));
+    expect(output.crawlRequests[1]).toEqual(crawlRequestServerToClient(crawlRequests[1]));
```

Comment:
[Not a blocker] I'm not a huge fan of these assertions. `output` is another fn (what does it do? I have no idea at a glance without having to look it up) so basically this reads to me as `foo(x)` = `bar(x)` and I have no reference to what either of the outputs look like.

I'm not sure if or why it's not realistic to do a static object equality here, but it would be really nice to see it.

Reply:
I did this in 6f7edbc

Reply:
Super appreciated!